elastic/kibana
# [Upgrade Assistant] Fix deprecated index settings resolution

Fixes https://github.com/elastic/kibana/issues/123501

This PR fixes the above bug and adds an API test to check the ES API returns the expected message. 

Ideally, we have a more robust solution in the long term. Elasticsearch did add a `meta` field to the deprecation info API via https://github.com/elastic/elasticsearch/issues/73089. Perhaps we can build upon this in the future. Relates to https://github.com/elastic/kibana/issues/119834.

### How to test
Run the following command in Console and verify the deprecation shows up in UA with the ability to remove the deprecated setting. There is also an example in the [UA readme](https://github.com/elastic/kibana/tree/7.17/x-pack/plugins/upgrade_assistant).

```
PUT deprecated_settings
{
  "settings": {
    "translog.retention.size": "1b",
    "translog.retention.age": "5m",
    "index.soft_deletes.enabled": true,
  }
}
```
